### PR TITLE
[WIP] Make Chosen framework agnostic using native JS

### PR DESCRIPTION
**Pull request by caccialdo**
This PR takes over previous efforts to make Chosen framework-agnostic:

* [Daniel Lo Nigro](https://github.com/harvesthq/chosen/pull/1394)
* [Arnaud Ceccaldi](https://github.com/harvesthq/chosen/pull/1727)

This native version is now up-to-date with `master` as of January 23rd, 2014.

### CODE REVIEW

**Comment by Daniel15** [comment]
Ping

What's the status of this?

**Comment by caccialdo** [comment]
I guess the changes since this pull request would need to get applied if the team wanted to get ahead with it.

**Comment by timeinfeldt** [comment]
Ping

What's the status of this?

**Comment by woodworker** [comment]
Ping

What's the status of this?

**Comment by Burtchen** [comment]
We have tested this extensively for quite a while, and it would be great to see the PR movig forward!

**Comment by Darijusch** [comment]
+1

File: coffee/chosen.native.coffee [hunk 1]
```diff
@@ -0,0 +1,512 @@
...
+    for key, value of container_props
+      @container.setAttribute key, value
+
+    if @is_multiple
+      @container.innerHTML = '<ul class="chosen-choices"><li class="search-field"><input type="text" value="' + @default_text + '" class="default" autocomplete="off" style="width:25px;" /></li></ul><div class="chosen-drop"><ul class="chosen-results"></ul></div>'
+    else
+      @container.innerHTML = '<a class="chosen-single chosen-default" tabindex="-1"><span>' + @default_text + '</span><div><b></b></div></a><div class="chosen-drop"><div class="chosen-search"><input type="text" autocomplete="off" /></div><ul class="chosen-results"></ul></div>'
+
+    @form_field.style.display = 'none'
+    @form_field.parentNode.appendChild @container
```
Comment (danez) [inline coffee/chosen.native.coffee]: This is not the same behavior as in the jquery or prototype version.

Should be:
```
@form_field.nextSibling ? @form_field.parentNode.insertBefore(@container, @form_field.nextSibling) : @form_field.parentNode.appendChild @container
```

**Comment by dgt41** [comment]
Any chances for this to be merged?

**Comment by zslabs** [comment]
Not sure how much of this is flushed out by now, but I'd recommend moving away from polyfills and allowing users to add-in these as they see fit. As a shameless plug, I [wrote about this](https://zslabs.com/articles/polyno) on a good alternative solution to mounds of polyfills in 3rd party scripts.

**Comment by Daniel15** [comment]
> but I'd recommend moving away from polyfills and allowing users to add-in these as they see fit

My original work was from 2013, when IE6 and IE7 support was more important and Chosen would have to work mostly standalone on a page. Things are a bit different now, we could probably remove all polyfill-like code from Chosen and just make people load the polyfills themselves if they want to support old browsers.

I've pretty much lost hope on something like this ever being merged. Between this and the accessibility issues in Chosen, I moved away from it for most sites. My previous employer wasn't able to use Chosen due to the accessibility issues.

**Comment by zslabs** [comment]
@Daniel15 Appreciate your thoughts around this. Not that this is the most appropriate place to be asking, but have you found a suitable replacement that is framework independent?

**Comment by Daniel15** [comment]
> have you found a suitable replacement that is framework independent

I'm using React quite a bit these days, so personally I like http://jedwatson.github.io/react-select/

**Comment by htstudios** [comment]
@Daniel15 looks good, however something that does not require ReactJS for conventional sites might be great... like Chosen without JQuery.
Any news on this?

**Comment by pfiller** [comment]
In an effort to try and get Chosen's repository under control and to best serve the people who use the tool, we're closing pull requests that add new functionality or make significant changes. For the time being, we're only merging bug fixes and functionality changes that are inline with Chosen's current scope. This doesn't mean we won't ever consider this functionality, but it does mean that we're being realistic about the time constraints the project maintainers have to give to this project.

Sorry this has sat open so long @caccialdo. The hope is that by getting things under control, we can get to a place in the future where we can once again start adding new functionality (at which point we'd come back to closed PRs like this). Thanks for your patience and understanding!

**Comment by inoas** [comment]
Time to move on to http://jedwatson.github.io/react-select/ to get rid of jQuery dependency - sadly.

**Comment by Daniel15** [comment]
Wow, I can't believe it's already been three years since I did the initial work on making Chosen framework-agnostic.

@pfiller - Is there any ETA for that, or a roadmap for Chosen? It's not really going to remain relevant if it continues to depend on jQuery and/or Prototype.

**Comment by pfiller** [comment]
Our roadmap is to create some breathing room so we can decide what to do next. Over the years, we've let too many PRs stay open. The feeling is that merging one creates a house of cards where every other pull request is suddenly un-mergeable. It's proven impossible to add new functionality or do significant refactoring when every pull request feels dependent on the other.

A Chosen that has no dependencies would be a nice thing to have, but thankfully a lot of great plugins have come along to serve the various needs of the modern JS toolkit. @inoas thinks it's sad that something like [react-select](http://jedwatson.github.io/react-select/) does something Chosen could be doing, but I think it's freaking neat! There are so many Chosen-like things out there now - why get hung up on what one project does?

I really do appreciate your efforts @Daniel15 even if we didn't show it by merging your code. We just need to be realistic about what's going to happen here going forward. When we have a plan, my hope is that we can publish it so people know how they can help us do whatever is next.

**Pull request closed without merging**